instrumentation/http-url-connection/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/httpurlconnection/HttpUrlConnectionInstrumentation.java

```diff
@@ -116,6 +116,11 @@ public static void methodExit(
       callDepth.getAndIncrement();
       try {
         scope.close();
+        Class<? extends HttpURLConnection> connectionClass = connection.getClass();
+
+        GetOutputStreamContext getOutputStreamContext =
```

Comment:
I'd use the value of `connection.getRequestMethod` here. If `getOutputStream` got an exception (maybe `setDoOutput(true)`) wasn't called then the request method isn't changed and we could still report it as `GET`.